README.md

```diff
@@ -1,7 +1,158 @@
 # FTAPIKit
 
-Declarative, generic REST API framework using URLSession and Codable
+![Swift](https://github.com/futuredapp/FTAPIKit/workflows/Swift/badge.svg)
+![Cocoapods](https://img.shields.io/cocoapods/v/FTAPIKit)
+![Cocoapods platforms](https://img.shields.io/cocoapods/p/FTAPIKit)
+![License](https://img.shields.io/cocoapods/l/FTAPIKit)
 
-## FTAPIKit templates installation
+Declarative and generic REST API framework using Codable.
+With standard implementation using URLSesssion and JSON encoder/decoder.
+Easily extensible for your asynchronous framework or networking stack.
 
-The templates can be added to Xcode simply running `make` inside `Templates/` directory.
+## Installation
+
+When using Swift package manager install using Xcode 11+
+or add following line to your dependencies:
+
+```swift
+.package(url: "https://github.com/futuredapp/FTAPIKit.git", from: "1.0.0")
+```
+
+When using CocoaPods add following line to your `Podfile`:
+
+```ruby
+pod 'FTAPIKit', '~> 1.0'
+```
+
+# Features
+
+The main feature of this library is to provide documentation-like API
+for defining web services. This is achieved using declarative
+and protocol-oriented programming in Swift.
+
+The framework provides two core protocols reflecting the physical infrastructure:
+
+- `Server` protocol defining single web service.
+- `Endpoint` protocol defining access points for resources.
+
+Combining instances of type conforming to `Server` and `Endpoint` we can build request.
+`URLServer` has convenience method for calling endpoints using `URLSession`.
+Recommended usage on application-side is implementing API client which encapsulates
+logic not provided by this framework like signing authorized endpoints
+or conforming to `URLSessionDelegate`.
+
+![Architecture](Documentation/Architecture.svg)
+
+There are many convenient `Endpoint` protocols for various use cases
+like multipart upload, automatic encoding/decoding.
+
+- `Endpoint` protocol body is empty.
+- `DataEndpoint` sends provided data in body.
+- `UploadEndpoint` sends file in body stream.
+- `MultipartEndpoint` combines body parts into body stream.
```

Comment:
I would suggest explaining this in more detail, I don't see a reason for avoiding longer sentence or more sentences. (Also I don't understand what do you mean by this :) )